bin/test-isolated

```diff
@@ -28,39 +36,72 @@ const resolveGitStatus = () =>
     }
   );
 
-const initialGitStatusDeferred = resolveGitStatus();
+const initialGitStatusDeferred = shouldApplyFsCleanupCheck ? resolveGitStatus() : null;
 
-const run = path => {
-  const onFinally = () =>
-    Promise.all([initialGitStatusDeferred, resolveGitStatus()]).then(
-      ([initialStatus, currentStatus]) => {
-        if (initialStatus !== currentStatus) {
-          process.stderr.write(chalk.red.bold(`${path} didn't clean created temporary files\n\n`));
-          process.exit(1);
-        }
-      }
-    );
-
-  return spawn('./bin/test', ['--require=sinon-bluebird', path], {
-    stdio: 'inherit',
-    env: { FORCE_COLOR: '1', PATH: process.env.PATH },
-  }).then(onFinally, error =>
-    onFinally(error).then(() => {
-      process.stderr.write(chalk.red.bold(`${path} failed\n\n`));
-      if (error.code === 2) process.exit(2);
-      throw error;
-    })
-  );
-};
+const initialSetupDeferred = shouldApplyFsCleanupCheck
+  ? initialGitStatusDeferred
+  : Promise.resolve();
```

Comment:
I totally agree, and I definitely would not try to mix Bluebird with native promises in framework modules (especially that it may affect long stack traces which become handy)

Still this is an outer stand alone script, that by no means is mixed with framework logic or even invokes framework logic (note tests are run in other processes released with `spawn`)

It just distributes test runs, and also already uses three dependencies (`p-lmit`, `globby`, `child-process-ext`) which output native promises.

Also at some point (the sooner the better) we should switch to native promises, so configuring it with Bluebird (still partially) will imply an unnecessary need of refactor here.

Therefore I think it's totally fine to stick to native promises in this context.